iceoryx_posh/source/roudi/port_manager.cpp

```diff
@@ -619,6 +622,41 @@ PortManager::acquirePublisherPortData(const capro::ServiceDescription& service,
     return maybePublisherPortData;
 }
 
+cxx::expected<PublisherPortRouDiType::MemberType_t*, PortPoolError>
+PortManager::acquirePublisherPortDataWithoutDiscovery(const capro::ServiceDescription& service,
```

Comment:
The function `acquirePublisherPortData` is identical to this one, except the one line with the discovery. Could you please remove the whole content of that function and replace it with:
```cpp
return acquirePublisherPortDataWithoutDiscovery(service,
                                                     publisherOptions,
                                                     runtimeName,
                                                     payloadDataSegmentMemoryManager,
                                                     portConfigInfo)
                                                     .and_then([](auto publisherPortData){ doDiscoveryForPublisherPort(PublisherPortRouDiType(publisherPortData)); });
```

Reply:
Yes, but needs slight adaptation due to `doDiscoveryForPublisherPort` requiring an lvalue.